service/routers/decks.js

```diff
@@ -58,31 +58,41 @@ const createCard = async (req, res) => {
 const deleteDeck = async (req, res) => {
   const userId = ''
   const deckId = req.params.id
-  try {
-    const user = await User.findById(userId)
-    const removedDeck = user.decks.id(deckId).remove()
-    console.log(removedDeck)
-    user.save()
-    res.sendStatus(204)
-  } catch (err) {
-    console.log(`${deleteDeck.name}: ${err}`)
-    res.sendStatus(500)
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor._id.toString() === req.params.id.toString()) {
+    try {
+      const user = await User.findById(userId)
```

Comment:
Are you guaranteed to find the deck in the user making the request? What about admins deleting decks for other users?